pkg/client/client.go

```diff
@@ -115,99 +187,64 @@ func (n *NutanixClientHelper) GetClientFromEnvironment(ctx context.Context, nuta
-	if cred.Username == "" {
-		return nil, fmt.Errorf("could not create client because username was not set")
+func buildClientFromCredentials(creds prismgoclient.Credentials, additionalTrustBundle string) (*nutanixClientV3.Client, error) {
+	if creds.Username == "" {
+		return nil, ErrPrismIUsernameNotSet
 	}
-	if cred.Password == "" {
-		return nil, fmt.Errorf("could not create client because password was not set")
+	if creds.Password == "" {
+		return nil, ErrPrismIPasswordNotSet
 	}
-	if cred.Port == "" {
-		cred.Port = defaultEndpointPort
+	if creds.Port == "" {
+		creds.Port = defaultEndpointPort
 	}
-	if cred.URL == "" {
-		cred.URL = fmt.Sprintf("%s:%s", cred.Endpoint, cred.Port)
+	if creds.URL == "" {
+		creds.URL = fmt.Sprintf("%s:%s", creds.Endpoint, creds.Port)
 	}
+
 	clientOpts := make([]nutanixClientV3.ClientOption, 0)
 	if additionalTrustBundle != "" {
 		clientOpts = append(clientOpts, nutanixClientV3.WithPEMEncodedCertBundle([]byte(additionalTrustBundle)))
 	}
-	cli, err := nutanixClientV3.NewV3Client(cred, clientOpts...)
+	// Build the client with the creds and possibly an additional TrustBundle
+	cli, err := nutanixClientV3.NewV3Client(creds, clientOpts...)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create new nutanix client: %w", err)
 	}
 
-	// Check if the client is working
-	_, err = cli.V3.GetCurrentLoggedInUser(context.Background())
-	if err != nil {
-		return nil, fmt.Errorf("failed to get current logged in user with client: %w", err)
-	}
 	return cli, nil
 }
 
-func (n *NutanixClientHelper) getManagerNutanixPrismEndpoint() (*credentialTypes.NutanixPrismEndpoint, error) {
-	npe := &credentialTypes.NutanixPrismEndpoint{}
-	config, err := n.readEndpointConfig()
+// readManagerNutanixPrismEndpointFromFile reads the config file and unmarshalls it into NutanixPrismEndpoint.
+// Returns an error if the file does not exist and other read or unmarshalling errors.
+func readManagerNutanixPrismEndpointFromFile() (*credentials.NutanixPrismEndpoint, error) {
+	// fail on all errors including NotFound error
+	config, err := os.ReadFile(configPath)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read config: %w", err)
+		return nil, fmt.Errorf("failed to read prism config in manager: %w", err)
 	}
+	npe := &credentials.NutanixPrismEndpoint{}
 	if err = json.Unmarshal(config, npe); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal config: %w", err)
 	}
 	if npe.CredentialRef == nil {
-		return nil, fmt.Errorf("credentialRef must be set on CAPX manager")
+		return nil, ErrCredentialRefNotSet
 	}
 	return npe, nil
 }
-
-func (n *NutanixClientHelper) readEndpointConfig() ([]byte, error) {
-	if b, err := os.ReadFile(filepath.Join(configPath, endpointKey)); err == nil {
-		return b, err
-	} else if os.IsNotExist(err) {
-		return []byte{}, nil
-	} else {
-		return []byte{}, err
-	}
-}
```

Comment:
Removed this and just replaced with `os.ReadFile` directly, this method was **not** failing with an empty file, but then `if npe.CredentialRef == nil {` would fail with an misleading error since the struct is just empty.
Instead this will now fail if the file is not found.